src/driver/amdxdna/npu_mailbox.c

```diff
@@ -613,6 +647,13 @@ int xdna_mailbox_destroy_channel(struct mailbox_channel *mb_chann)
 	list_del(&mb_chann->chann_entry);
 	mutex_unlock(&mb_chann->mb->mbox_lock);
 
+	/* Disalbe an irq and wait. This might sleep. */
+	disable_irq(mb_chann->msix_irq);
+
+	/* Cancel RX work and wait for it to finish */
+	cancel_work_sync(&mb_chann->rx_work);
```

Comment:
I'm OK with taking disable/cancel type of action, but not comfortable with doing "destroy" and "free", before the context is completed removed from device. Can we split them and only do the "disable" and "cancel" part upfront?